src/rotation.zig

```diff
@@ -321,6 +321,190 @@ pub const RotationModes = struct {
+			}
+			branchModels.put(modelId, modelIndex) catch unreachable;
+			return modelIndex;
+		}
+
+		pub fn model(block: Block) u16 {
+			return blocks.meshes.modelIndexStart(block) + (block.data & 63);
+		}
+
+		pub fn generateData(
+			_: *main.game.World,
+			_: Vec3i,
+			_: Vec3f,
+			_: Vec3f,
+			_: Vec3i,
+			neighbor: ?Neighbor,
+			currentBlock: *Block,
+			neighborBlock: Block,
+			blockPlacing: bool,
+		) bool {
+			const blockBaseModel = blocks.meshes.modelIndexStart(currentBlock.*);
+			const neighborBaseModel = blocks.meshes.modelIndexStart(neighborBlock);
+
+			if(blockPlacing or (blockBaseModel == neighborBaseModel) or neighborBlock.solid()) {
+				const neighborModel = blocks.meshes.model(neighborBlock);
+
+				var currentData = BranchData.init(currentBlock.data);
+				// Branch block upon placement should extend towards a block it was placed
+				// on if the block is solid or also uses branch model.
+				const targetVal = (
+					(neighborBlock.solid() and !neighborBlock.viewThrough())
+					and ((blockBaseModel == neighborBaseModel) or main.models.models.items[neighborModel].isNeighborOccluded[neighbor.?.reverse().toInt()])
+				);
+				currentData.setConnection(neighbor.?, targetVal);
+
+				const result: u16 = currentData.enabledConnections;
+				if(result == currentBlock.data) return false;
+
+				currentBlock.data = result;
+				return true;
+			}
+			return false;
+		}
+
+		pub fn updateData(block: *Block, neighbor: Neighbor, neighborBlock: Block) bool {
+			const blockBaseModel = blocks.meshes.modelIndexStart(block.*);
+			const neighborBaseModel = blocks.meshes.modelIndexStart(neighborBlock);
+			var currentData = BranchData.init(block.data);
+
+			// Handle joining with other branches. While placed, branches extend in a
+			// opposite direction than they were placed from, effectively connecting
+			// to the block they were placed at.
+			if(blockBaseModel == neighborBaseModel) {
+				const neighborData = BranchData.init(neighborBlock.data);
+				currentData.setConnection(neighbor, neighborData.isConnected(neighbor.reverse()));
+			} else if(!neighborBlock.solid()) {
+				currentData.setConnection(neighbor, false);
+			}
+
+			const result: u16 = currentData.enabledConnections;
+			if(result == block.data) return false;
+
+			block.data = result;
+			return true;
+		}
+
+		fn closestRay(
+			block: Block,
+			relativePlayerPos: Vec3f,
+			playerDir: Vec3f
+		) ?u16 {
+			var resultIntersection: ?RayIntersectionResult = null;
+			var resultBitMask: ?u16 = null;
+			// Check intersection with central part of the branch block.
+			{
+				const modelIndex = blocks.meshes.modelIndexStart(block);
+				if(RotationMode.DefaultFunctions.rayModelIntersection(modelIndex, relativePlayerPos, playerDir)) |intersection| {
+					// We can't return here, as any intersection that accidentally passes
+					// through the central part would delete the whole block, even if player
+					// was aiming at the connection.
```

Comment:
I don't like that this function now has more comments than actual code.
This one here for example should have been just a review comment on my request to use early returns.